app-server/app/src/main/scala/organizations/Organizations.scala

```diff
@@ -0,0 +1,328 @@
+  )(
+    implicit
+    database: Database, ec: ExecutionContext
+  ): Query[UsersToOrganizations, UsersToOrganizations#TableElementType, Seq] = {
+    import database.driver.api._
+
+    sortMap.headOption match {
+      case Some(("id", order)) =>
+        order match {
+          case Order.Asc => applyUserRoleSort(query.sortBy(_.userId.asc), sortMap.tail)
+          case Order.Desc => applyUserRoleSort(query.sortBy(_.userId.desc), sortMap.tail)
+        }
+      case Some(("role", order)) =>
+        order match {
+          case Order.Asc => applyUserRoleSort(query.sortBy(_.role.asc), sortMap.tail)
+          case Order.Desc => applyUserRoleSort(query.sortBy(_.role.desc), sortMap.tail)
+        }
+      case Some(("modified", order)) =>
+        order match {
+          case Order.Asc => applyUserRoleSort(query.sortBy(_.modifiedAt.asc), sortMap.tail)
+          case Order.Desc => applyUserRoleSort(query.sortBy(_.modifiedAt.desc), sortMap.tail)
+        }
+      case Some(("created", order)) =>
+        order match {
+          case Order.Asc => applyUserRoleSort(query.sortBy(_.createdAt.asc), sortMap.tail)
+          case Order.Desc => applyUserRoleSort(query.sortBy(_.createdAt.desc), sortMap.tail)
+        }
+      case Some((_, order)) => applyUserRoleSort(query, sortMap.tail)
+      case _ => query
+    }
+  }
+
+  def getOrganizationList(page: PageRequest)(
+    implicit
+    database: Database, ec: ExecutionContext
+  ): Future[PaginatedResponse[OrganizationsRow]] = {
+    import database.driver.api._
+
+    val organizationsQueryResult = database.db.run {
+      applyOrgSort(Organizations, page.sort)
+        .drop(page.offset * page.limit)
+        .take(page.limit)
+        .result
+    }
+    val totalOrganizationsQuery = database.db.run {
+      Organizations.length.result
+    }
+
+    for {
+      totalOrganizations <- totalOrganizationsQuery
+      organizations <- organizationsQueryResult
+    } yield {
+      val hasNext = (page.offset + 1) * page.limit < totalOrganizations // 0 indexed page offset
+      val hasPrevious = page.offset > 0
+      PaginatedResponse(totalOrganizations, hasPrevious, hasNext,
+        page.offset, page.limit, organizations)
+    }
+  }
+
+  def getOrganization(id: java.util.UUID)(implicit database: Database):
+      Future[Option[OrganizationsRow]] = {
+    import database.driver.api._
+
+    database.db.run {
+      Organizations.filter(_.id === id).result.headOption
+    }
+  }
+
+  def createOrganization(org: OrganizationsRowCreate)(
+    implicit
+    database: Database, ec: ExecutionContext
+  ): Future[Try[OrganizationsRow]] = {
+    import database.driver.api._
+
+    val rowInsert = org.toOrganizationsRow()
+
+    database.db.run {
+      Organizations.forceInsert(rowInsert).asTry
+    } map {
+      case Success(res) => {
+        res match {
+          case 1 => Success(rowInsert)
+          case _ => Failure(new Exception("Unexpected result from database when inserting organization"))
```

Comment:
Probably should include the result in the exception to make it easier to identify a problem.